core/src/test/java/ai/timefold/solver/core/impl/domain/solution/cloner/AbstractSolutionClonerTest.java

```diff
@@ -1050,51 +1060,51 @@ private void assertDeepCloningEntityClone(TestdataFieldAnnotatedDeepCloningEntit
+        var cloneShadowVariableList = cloneEntity.getShadowVariableList();
         if (originalShadowVariableList == null) {
             assertThat(cloneShadowVariableList).isNull();
         } else {
             assertThat(cloneShadowVariableList).isNotSameAs(originalShadowVariableList);
             assertThat(cloneShadowVariableList).hasSameSizeAs(originalShadowVariableList);
-            for (int i = 0; i < originalShadowVariableList.size(); i++) {
+            for (var i = 0; i < originalShadowVariableList.size(); i++) {
                 assertThat(cloneShadowVariableList.get(i)).isSameAs(originalShadowVariableList.get(i));
             }
         }
 
-        Map<String, String> originalShadowVariableMap = originalEntity.getShadowVariableMap();
-        Map<String, String> cloneShadowVariableMap = cloneEntity.getShadowVariableMap();
+        var originalShadowVariableMap = originalEntity.getShadowVariableMap();
+        var cloneShadowVariableMap = cloneEntity.getShadowVariableMap();
         if (originalShadowVariableMap == null) {
             assertThat(cloneShadowVariableMap).isNull();
         } else {
             assertThat(cloneShadowVariableMap).isNotSameAs(originalShadowVariableMap);
             assertThat(cloneShadowVariableMap).hasSameSizeAs(originalShadowVariableMap);
-            for (String key : originalShadowVariableMap.keySet()) {
+            for (var key : originalShadowVariableMap.keySet()) {
                 assertThat(cloneShadowVariableMap.get(key)).isSameAs(originalShadowVariableMap.get(key));
             }
         }
 
-        Map<List<String>, String> originalStringListToStringMap = originalEntity.getStringListToStringMap();
-        Map<List<String>, String> cloneStringListToStringMap = cloneEntity.getStringListToStringMap();
+        var originalStringListToStringMap = originalEntity.getStringListToStringMap();
+        var cloneStringListToStringMap = cloneEntity.getStringListToStringMap();
         if (originalStringListToStringMap == null) {
             assertThat(cloneStringListToStringMap).isNull();
         } else {
             assertThat(cloneStringListToStringMap).isNotSameAs(originalStringListToStringMap);
             assertThat(cloneStringListToStringMap).hasSameSizeAs(originalStringListToStringMap);
-            for (List<String> key : originalStringListToStringMap.keySet()) {
+            for (var key : originalStringListToStringMap.keySet()) {
                 assertThat(cloneStringListToStringMap.get(key)).isSameAs(originalStringListToStringMap.get(key));
             }
         }
 
-        Map<String, List<String>> originalStringToStringListMap = originalEntity.getStringToStringListMap();
-        Map<String, List<String>> cloneStringToStringListMap = cloneEntity.getStringToStringListMap();
+        var originalStringToStringListMap = originalEntity.getStringToStringListMap();
+        var cloneStringToStringListMap = cloneEntity.getStringToStringListMap();
```

Comment:
IMO we need to come up with some sort of a guideline for when we make these `var` refactors.
This PR has 800 affected files. It is near impossible to verify the main changes, if they're intertwined with these additional `var` changes as well.

I have done this in the past too, I'm not assigning any blame. But going forward, I'm thinking we'll all have to have more discipline to only stick to the related changes.

Reply:
Makes sense. Should I revert the `var` changes?

Reply:
No need anymore. I already went through all of it. But in the future, let's try to be cautious about this, on large PRs.